### PR TITLE
[docs,dv,pwm] Docs, DIF and DV corrections for PWM

### PR DESCRIPTION
**Pull request by alees24**
This PR collects a number of small, hopefully uncontentious corrections and modifications to the documentation, DIF and DV relating to PWM. These have been identified in the push towards D3.

### CODE REVIEW

**Review by rswarbrick — COMMENTED** [review]
Some nitty comments but this is a definite improvement. Thank you very much!

File: sw/device/lib/dif/dif_pwm.c [hunk 1]
```diff
@@ -99,7 +99,10 @@ dif_result_t dif_pwm_configure_channel(const dif_pwm_t *pwm,
   if (config.mode == kDifPwmModeHeartbeat) {
     param_reg =
         bitfield_bit32_write(param_reg, PWM_PWM_PARAM_0_HTBT_EN_0_BIT, true);
-  } else if (config.mode == kDifPwmModeBlink) {
+  }
+  // Blink behavior is modulated by the heartbeat enable bit and must still be
+  // set.
```
[the remark this answers is not in the document]
Comment (alees24) [reply sw/device/lib/dif/dif_pwm.c:104]: Engineering jargon; it's semantically correct. The heartbeat enable bit changes ('modulates') the behaviour of blink mode...I'll reword it, though.

File: hw/ip/pwm/doc/registers.md [hunk 2]
```diff
@@ -58,10 +58,10 @@ Register write enable for all control registers
...
+|  Bits  |  Type  |  Reset  | Name   | Description                                                                                                                                                                                                                           |
+|:------:|:------:|:-------:|:-------|:--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+|  31:1  |        |         |        | Reserved                                                                                                                                                                                                                              |
+|   0    |  rw0c  |   0x1   | REGWEN | When true, all writable registers can be modified. When false, they become read-only. Defaults true, write zero to clear. This can be cleared after initial configuration at boot in order to lock in the supplied register settings. |
```
[the remark this answers is not in the document]
Comment (rswarbrick) [reply hw/ip/pwm/doc/registers.md:64]: Would you be up for squashing this with the first commit? (That would avoid having a "partial state" in the history)

**Comment by rswarbrick** [comment]
I think we should probably squash the autogen commit, but then this looks good to me.

**Comment by alees24** [comment]
> I think we should probably squash the autogen commit, but then this looks good to me.

Done.

**Review by rswarbrick — APPROVED** [review]
Looks good to me: thanks!

**Comment by rswarbrick** [comment]
CHANGE AUTHORIZED: hw/ip/pwm/data/pwm.hjson

The change here is just fixing typos in documentation comments: definitely no risk to the design :-)

**Comment by alees24** [comment]
CHANGE AUTHORIZED: hw/ip/pwm/data/pwm.hjson

**Pull request merged**